source/vsm/vsm/db/sqlalchemy/api.py

```diff
@@ -4305,21 +4287,14 @@ def get_poolusage(context, poolusage_id):
     return result
 
 def latency_all_types(context, search_opts, session=None):#for latency
+    # Get diamond collection interval
```

Comment:
Essentially, the refactoring I've already done replaced this:
`
    metrics_name = search_opts['metrics_name']
    timestamp_start = search_opts.has_key('timestamp_start') and int(search_opts['timestamp_start']) or None
    timestamp_end = search_opts.has_key('timestamp_end') and int(search_opts['timestamp_end']) or None
    setting_ref = vsm_settings_get_by_name(context, 'cpu_diamond_collect_interval', session=session)
    if setting_ref:
        diamond_collect_interval = int(setting_ref['value'])
    else:
        diamond_collect_interval = 15
        vsm_settings_update_or_create(context, {'name':'cpu_diamond_collect_interval','value':diamond_collect_interval}, session=session)
    if timestamp_start is None and timestamp_end:
        timestamp_start = timestamp_end - diamond_collect_interval
    elif timestamp_start  and  timestamp_end is None:
        timestamp_start = timestamp_start + diamond_collect_interval
    if timestamp_start > timestamp_end:
        return []
`
with this:
`
    # Get diamond collection interval
    diamond_collect_interval = get_diamond_interval_value('cpu_diamond_collect_interval', context, session=session)
    if diamond_collect_interval == 0:
        return []
    # Get timestamps for metric interval
    timestamp_start, timestamp_end = get_metric_timestamp_range(context, search_opts, diamond_collect_interval)
    if timestamp_start > timestamp_end:
        return []
`